charmcraft.yaml

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+type: charm
+bases:
+  - build-on:
+      - name: "ubuntu"
+        channel: "22.04"
+    run-on:
+      - name: "ubuntu"
+        channel: "22.04"
+parts:
+  charm:
+    charm-binary-python-packages:
+      - jsonschema
+      - cryptography
```

Comment:
We use the binary packages as we need to install cryptography and jsonschema before other requirements and regular python packages, otherwise packing the charm fails (see https://github.com/canonical/oathkeeper-operator/actions/runs/8723861613/job/23936305235#step:4:1274)

Reply:
@natalian98 could you please try to add `pkg-config` to your `build-packages` in charmcraft.py and retry (after removing the `charm-binary-python-packages`

i was able to replicate the issue locally, and fix it with the added build package. 

also, reminder: please remove `charm-binary-python-packages` from the test charm `auth-proxy-requirer`'s charmcraft.py file

Reply:
When we faced the issue with rust, this was the suggested company fix. Other published charms install these libraries like this as well (eg https://github.com/canonical/grafana-k8s-operator/blob/main/charmcraft.yaml#L113). I think that changing it would create unnecessary variance.

Unless something has changed and this is how charm teams are going to install these packages from now on, I would suggest to leave it as it is.

Reply:
I am unsure that all teams are adopting the fix where cryptography is installed with `charm-binary-python-packages`. The data platform team, for instance, is adding `pkg-config` as a build package to resolve the rust error.

My intention was to raise the security concern, and in my opinion, it is perfectly acceptable to choose `charm-binary-python-packages` given that you are aware of the tradeoffs